### PR TITLE
docs: guideline for fork testing

### LINKED ISSUE #715: Document the GitHub secrets required to make the CI work in PRs
We should update the [`CONTRIBUTING`](https://github.com/sablier-labs/v2-core/blob/main/CONTRIBUTING.md) guide to explain what GitHub secrets contributors need to have set in order to make the CI pass in their forks.

Ref https://github.com/sablier-labs/v2-core/pull/714#issuecomment-1776954027

### PR DESCRIPTION
**Pull request by smol-ninja**
fixes #715

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -52,10 +53,20 @@ When making a pull request, ensure that:
...
-the appropriate environment values. You need to provide your mnemonic phrase and a few API keys.
+To make CI work in your pull request, ensure that the necessary environment variables are configured in your forked
+repository's secrets. Please add the following variables to your
+[github secret](https://docs.github.com/en/actions/security-guides/using-secrets-in-github-actions):
+
+- RPC_URL_MAINNET
+- API_KEY_INFURA
+- API_KEY_ETHERSCAN
+
+### Local setup
```
Comment (PaulRBerg) [inline CONTRIBUTING.md:66]: Let's make this the first section.

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -52,10 +53,20 @@ When making a pull request, ensure that:
...
 ## Environment Variables
 
-Some of the features of this repository, such as deployments, require environment variables to be set up.
+### Deployment
 
-Follow the [`.env.example`](./.env.example) file to create a `.env` file at the root of the repo and populate it with
-the appropriate environment values. You need to provide your mnemonic phrase and a few API keys.
+To make CI work in your pull request, ensure that the necessary environment variables are configured in your forked
+repository's secrets. Please add the following variables to your
+[github secret](https://docs.github.com/en/actions/security-guides/using-secrets-in-github-actions):
```
Comment (PaulRBerg) [inline CONTRIBUTING.md:60]: Let's say "in your GitHub Secrets".

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -52,10 +53,20 @@ When making a pull request, ensure that:
...
+
+- RPC_URL_MAINNET
```
Comment (PaulRBerg) [inline CONTRIBUTING.md:62]: Let's order this list alphabetically

**Review by PaulRBerg — CHANGES_REQUESTED** [review]
Thanks @smol-ninja, generally LGTM

**Comment by PaulRBerg** [comment]
Btw I suppose that you created this PR from your personal fork to test the environment variables, but for future PRs, you should be able to create branches directly in the canonical V2 Core repo.

**Comment by smol-ninja** [comment]
> Btw I suppose that you created this PR from your personal fork to test the environment variables, but for future PRs, you should be able to create branches directly in the canonical V2 Core repo.

Yes, you got it. But the issue is not yet resolved. The fork tests will fail on a PR requested by an external contributor despite them passing in the local repo.

Secrets are not passed from the source branch to the workflows during a PR raised by an external contributor.

Ref: https://docs.github.com/en/actions/security-guides/using-secrets-in-github-actions#using-secrets-in-a-workflow
> With the exception of GITHUB_TOKEN, secrets are not passed to the runner when a workflow is triggered from a forked repository

There is a solution: by using `pull_request_target` instead of `pull_request` event in GitHub Actions, the workflow will operate within the context of the contributor's forked repository secrets each time they open a new pull request.

Ref:
- https://docs.github.com/en/actions/managing-workflow-runs/approving-workflow-runs-from-public-forks#about-workflow-runs-from-public-forks
- https://github.blog/2020-08-03-github-actions-improvements-for-fork-and-pull-request-workflows/
- https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows#pull_request_target

Risks involved:
We just have to be extra careful when using `pull_request_target` as mentioned in this [Github blog](https://securitylab.github.com/research/github-actions-preventing-pwn-requests/#:%7E:text=of%20the%20PR.-,Having%20said%20that,-%2C%20mixing%20pull_request_target%20with).

Should I create another issue to look at it in the future if its not a priority? I can also go ahead and do some local testing in my repo to see how `pull_request_target` event works.

**Comment by smol-ninja** [comment]
@PaulRBerg should I merge it?

**Comment by PaulRBerg** [comment]
Apologies @smol-ninja, it's a busy day on the business front. I will review this PR later, if that's ok.

**Comment by PaulRBerg** [comment]
> the issue is not yet resolved. The fork tests will fail on a PR requested by an external contributor despite them passing in the local repo.

I see!

> Should I create another issue to look at it in the future if its not a priority?

Yes, please.

> should I merge it?

Merged now. Thank you for your contribution.

**Pull request merged**